src/BenchmarksApps/Kestrel/PlatformBenchmarks/SocketPipes/SocketAwaitableEventArgs.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PlatformBenchmarks
+{
+    // more or less a copy of https://github.com/dotnet/aspnetcore/blob/master/src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketAwaitableEventArgs.cs
+    // but without PipeScheduler (always inlining)
+    internal sealed class SocketAwaitableEventArgs : SocketAsyncEventArgs, ICriticalNotifyCompletion
```

Comment:
Do you need this at all? You can use the SocketTask APIs that do this internally. Kestrel only does this because it's trying to schedule directly to the IOQueue from continuations.

Reply:
The cost of awaiting `socket.ReadAsync` is quite big, by using this custom class I was able  to get +15k RPS. I am going to create an issue in runtime repo

Reply:
cc @stephentoub

Reply:
> The cost of awaiting socket.ReadAsync is quite big, by using this custom class I was able to get +15k RPS.

"quite big" is ~15K out of ~1065k in a benchmark that's basically seeing how fast it can call this method?  To me that's not worth all the extra code and complexity and diverging from how we want developers to use Socket.

> I am going to create an issue in runtime repo

Recommending what?  If you have concrete ideas for improving it, great, but it has to account for a bunch of things you're not here, like the possibility it might be used concurrently, being used via an interface via ValueTask, that there might be a custom SynchronizationContext/TaskScheduler to be used, etc.